pycaret/internal/tabular.py

```diff
@@ -6643,11 +6659,21 @@ def confusion_matrix():
 
             from yellowbrick.classifier import ConfusionMatrix
 
+            try:
+                dtype_object = get_config('prep_pipe').steps[0][1]
+                encoder_labels = dtype_object.le.transform(dtype_object.le.classes_)
+                encoder_classes = dtype_object.le.classes_
+                encoder_dictionary = dict(zip(encoder_labels,encoder_classes))
+            except:
+                encoder_dictionary = None
+
             visualizer = ConfusionMatrix(
                 pipeline_with_model,
                 random_state=seed,
                 fontsize=15,
                 cmap="Greens",
+                encoder=encoder_dictionary,
+                **plot_kwargs
```

Comment:
shouldn't this be added to other visualizers as well?

Reply:
Yes. The PR is not complete yet. I will add it to all visualizers.